### DISCUSSION TITLE
keys-pressed, keys-to-press highlighting?

### DISCUSSION THREAD

**Discussion by jackpipe** [ggandor/leap.nvim#161] [General]
This is a great plugin thank you, and really seems to have potential, but I admit I'm quite confused by the way highlighting works or is supposed to work.

I have

```
lua require('leap').opts.highlight_unlabeled = true
lua require('leap').opts.highlight_unlabeled_phase_one_targets = true
```

So when I type the first letter, I get two sets of things highlighted:
- destinations where I only need to type the second letter. Both the currently typed letter, and the next one I need to type are highlighted in the same color.
- 3rd-letter-to-type labels. The 3rd-letter/label is highlighted in a different color, but not the first two letters.

And so what I keep end up doing is typing the first letter, and then the highlighted label, forgetting I have to type the second letter (because it's not highlighted).

What I'd like is for there to be 'progressive match' highlighting in two different colors, one showing the already-entered character matches, and the second showing the upcoming-letter matches.

So let's say my colors are pink and yellow.

I type 'f' 'p' to match all 'p's forwards. All 'p's will be highlighted in pink, followed by 1 or two letters (an actual next character, and then a label if required') highlighted in yellow.

Now I type 'a', and all 'pa' matches are highlighted in pink, followed by a yellow-highlighted label for all but the first instance of 'pa'.

Is that possible with the current config options, or an easy hack?
If not is there a way to at least ensure the first two characters are highlighted, even when they have a label, not only if they are 'standalone' targets?

**Comment by ggandor** [comment]
> Is that possible with the current config options, or an easy hack?

Not at the moment.

> And so what I keep end up doing is typing the first letter, and then the highlighted label, forgetting I have to type the second letter (because it's not highlighted).
 
https://github.com/ggandor/leap.nvim/issues/65#issuecomment-1282971198

> If not is there a way to at least ensure the first two characters are highlighted, even when they have a label, not only if they are 'standalone' targets?

#120 (but see #65)
